src/App.tsx

```diff
@@ -21,75 +22,147 @@ enum SortType {
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
+//   // Sort and reverse goods if needed
+//   if (isReversed) {
+//     visibleGoods.reverse()
+//   }
+
+//   visibleGoods.sort((good1: string, good2: string) => {
+//     switch (sortType) {
+//       case SortType.ALPHABET:
+//         return good1.localeCompare(good2)
+
+//       case SortType.LENGTH:
+//         return good1.length - good2.length
+
+//       default:
+//         return 0
+//     }
+//   })
+
+//   // eslint-disable-next-line no-console
+//   console.log(sortType, isReversed);
+
+//   return visibleGoods;
 // };
 
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
 
-      <ul>
+export class App extends React.Component<{}, State> {
+  state: State = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  reverse = () => {
+    this.setState((state) => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sortByAlphapet = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  reset = () => {
+    this.setState({ sortType: SortType.NONE, isReversed: false });
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+
+    const visibleGoods = [...goodsFromServer];
+
+    visibleGoods.sort((good1: string, good2: string) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return (
+      <div className="section content box">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames(
+              'button is-info', { 'is-light': sortType !== SortType.ALPHABET },
+            )}
+            onClick={this.sortByAlphapet}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button is-succes', { 'is-light': sortType !== SortType.LENGTH },
```

Comment:
you misspelled `is-success` class name, therefore your ui doesnt work properly
```suggestion
              'button is-success', { 'is-light': sortType !== SortType.LENGTH },
```